packages/dialog/__tests__/dialog.spec.ts

```diff
@@ -0,0 +1,187 @@
+
+  test('dialog should have a footer when footer has been given', () => {
+    const wrapper = _mount({
+      slots: {
+        footer: AXIOM,
+      },
+    })
+
+    expect(wrapper.find('.el-dialog__footer').exists()).toBe(true)
+    expect(wrapper.find('.el-dialog__footer').text()).toBe(AXIOM)
+  })
+
+  test('should append dialog to body when appendToBody is true', () => {
+    const wrapper = _mount({
+      props: {
+        appendToBody: true,
+      },
+    })
+    expect(document.body.firstElementChild.classList.contains('el-overlay')).toBe(true)
+    wrapper.unmount()
+  })
+
+  test('should center dialog', () => {
+    const wrapper = _mount({
+      props: {
+        center: true,
+      },
+    })
+    expect(wrapper.find('.el-dialog__center').exists()).toBe(true)
+  })
+
+  test('should show close button', () => {
+    const wrapper = _mount({})
+    expect(wrapper.find('.el-dialog__close').exists()).toBe(true)
+  })
+
+  test('should close dialog when click on close button', async () => {
+    const wrapper = _mount({
+      props: {
+        modelValue: true,
+      },
+    })
+
+    await wrapper.find('.el-dialog__headerbtn').trigger('click')
+    expect(wrapper.vm.visible).toBe(false)
+  })
+
+  describe('mask related', () => {
+    test('should not have overlay mask when mask is false', () => {
+      const wrapper = _mount({
+        props: {
+          mask: false,
+        },
+      })
+
+      expect(wrapper.find('.el-overlay').exists()).toBe(false)
+    })
+
+    test('should close the modal when clicking on mask when `closeOnClickModal` is true', async () => {
+      const wrapper = _mount({})
+
+      expect(wrapper.find('.el-overlay').exists()).toBe(true)
+
+      await wrapper.find('.el-overlay').trigger('click')
+      expect(wrapper.vm.visible).toBe(false)
+    })
+  })
+
+  describe('life cycles', () => {
+    test('should call before close', async () => {
+      const beforeClose = jest.fn()
+      const wrapper = _mount({
+        props: {
+          beforeClose,
+        },
+      })
+
+      wrapper.vm.handleClose()
+      expect(beforeClose).toHaveBeenCalled()
+    })
+
+    test('should not close dialog when user cancelled', () => {
+      const beforeClose = jest.fn().mockImplementation((hide: (cancel: boolean) => void) => hide(true))
+
+      const wrapper = _mount({
+        props: {
+          beforeClose,
+          modelValue: true,
+        },
+      })
+
+      wrapper.vm.handleClose()
+      expect(beforeClose).toHaveBeenCalled()
+      expect(wrapper.vm.visible).toBe(true)
+    })
+
+    test('should open and close with delay', async () => {
+      const wrapper = _mount({
+        props: {
+          openDelay: 200,
+          closeDelay: 200,
+          modelValue: false,
+        },
+      })
+
+      expect(wrapper.vm.visible).toBe(false)
+
+      await wrapper.setProps({
+        modelValue: true,
+      })
+
+      expect(wrapper.vm.visible).toBe(false)
+
+      jest.runOnlyPendingTimers()
```

Comment:
Not at all, since `setTimeout` API is provided by the browser itself, it should be implemented correctly which should always be correct when I actually running those code. That to be said `global.setTimeout` should always be a standard that invokes callback that I passed to it.
Using `runOnlyPendingTimers` saves time for the testing (because `setTimeout` should work as expected with specified delay, it should be irrelevant to my code (A.K.A. de-coupled).